expression/builtin_math_vec.go

```diff
@@ -697,11 +697,39 @@ func (b *builtinCeilIntToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chun
 }
 
 func (b *builtinTruncateIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinTruncateIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	if err := b.args[0].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	buf64s := buf.Int64s()
+
+	for i := 0; i < len(i64s); i++ {
+		if result.IsNull(i) {
+			i64s[i] = 0
```

Comment:
Maybe we can delete this line. Because when the ```result.IsNull(i)```, we won't visit i64[i]

Reply:
oh, I saw the unvec function, it returns 0 if the result is null, so I think we should make it 0. All right, I will delete this line.